shapeshift/web
# feat: display Yearn and uniV2 transactions via generic contract handling component

## Description

- Adds a generic `TransactionContract` component that handles contract interaction transactions, including Yearn vault and UniV2 liquidity interactions.
- Makes required package bumps

## Future improvements

- It'll require additional data from the unchained parser, but next, I'd like to add the contract that the transaction interacted within the expanded view.
- Add ZRX ABI and better support associated transactions
- Add a generic, fallback, transaction renderer

## Notes for the reviewer

- I'm checking with the product team to confirm the final UI
- Has a dependency on https://github.com/shapeshift/lib/pull/405

## Notice

Before submitting a pull request, please make sure you have answered the following:


## Pull Request Type


## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/1068

## Risk

Minimal - it doesn't touch existing code, and at worst, the new functionality won't work as promised.

## Testing

- Check that UniSwap add and remove liquidity transactions display
- Check that Yearn approve, deposit, and withdraw transactions display

## Screenshots (if applicable)

<img width="330" alt="Screen Shot 2022-02-28 at 4 41 20 pm" src="https://user-images.githubusercontent.com/97164662/155931086-fc9c8dcd-1154-46ed-aefe-83b748942ce4.png">
<img width="693" alt="Screen Shot 2022-02-28 at 4 41 30 pm" src="https://user-images.githubusercontent.com/97164662/155931094-088aab53-4322-425e-92b5-a25f8fef99d7.png">
<img width="689" alt="Screen Shot 2022-02-28 at 4 42 24 pm" src="https://user-images.githubusercontent.com/97164662/155931098-a444bef9-40ff-4f67-b75f-e5d9070c602f.png">
